RobotNavigation/Maze.cs

```diff
@@ -34,41 +27,19 @@ public Maze(int height, int width)
 
         public void Init()
         {
-            _maze = new Position.Cell[Width, Height];
+            _maze = new Position.CellType[Width, Height];
             for (int i = 0; i < Width; i++)
             {
                 for (int j = 0; j < Height; j++)
                 {
-                    _maze[i, j] = Position.Cell.Empty;
+                    _maze[i, j] = Position.CellType.Empty;
```

Comment:
_maze[i,j].X = i;
_maze[i,j].Y = j;
_maze[i,j].Cell = Position.CellType.Empty;

Reply:
I would also consider changing the name of the property Position.Cell to Position.Type instead.